packages/eslint-plugin/src/rules/init-declarations.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+} from '@typescript-eslint/experimental-utils';
+import baseRule from 'eslint/lib/rules/init-declarations';
+import {
+  InferOptionsTypeFromRule,
+  InferMessageIdsTypeFromRule,
+  createRule,
+  deepMerge,
+} from '../util';
+
+export type Options = InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
+
+const schema = deepMerge(
+  Array.isArray(baseRule.meta.schema)
+    ? baseRule.meta.schema[0]
+    : baseRule.meta.schema,
+);
+
+export default createRule<Options, MessageIds>({
+  name: 'init-declarations',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'require or disallow initialization in variable declarations',
+      category: 'Variables',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: ['always'],
+  create(context) {
+    const rules = baseRule.create(context);
+    const mode = context.options[0] || 'always';
+
+    return {
+      'VariableDeclaration:exit'(
+        node: TSESTree.VariableDeclaration | TSESTree.Node,
```

Comment:
@bradzacher for both 

`TSESTree.VariableDeclaration | TSESTree.Node`

and

`TSESTree.VariableDeclaration | ts.Node`

The same errors are coming about the `declare` not found. `VariableDeclaration` has that but still, I dont know why its not able to locate it.

Reply:
Should I create a custom type for this node ? it seems `VariableDeclaration` and `Node` doesn't have common type fields like `parent` and `declare`

Reply:
`TSESTree.VariableDeclaration | TSESTree.Node` is the incorrect type.
`TSESTree.Node` is the union of all the node types, so `TSESTree.VariableDeclaration | TSESTree.Node === TSESTree.Node`.

`parent` is always optional due to how the parser works - this is intentional.
```suggestion
        node: TSESTree.VariableDeclaration,
```